commits.webkit.org should have a UI to inter-convert revision/hash/commit-identifier

briannafan opened the pull request:
#### 95562dee879669cfde6fb314b7399494464ca759
<pre>
commits.webkit.org should have a UI to inter-convert revision/hash/commit-identifier
<a href="https://bugs.webkit.org/show_bug.cgi?id=225540">https://bugs.webkit.org/show_bug.cgi?id=225540</a>
<a href="https://rdar.apple.com/78034339">rdar://78034339</a>

Reviewed by Jonathan Bedard.

Introduces UI to commits.webkit.org!
Search and compare commits.

* Tools/Scripts/libraries/reporelaypy/reporelaypy/static/css/webkit.css: Added.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/main.js: Added.
(SetUpInputs):
* Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/search.js: Added.
(SetUpInputs):
(ErrorMessage):
(async GetCommit):
(ToggleButtons):
(ValidateCommitFormat):
(async GenerateCompareLink):
(async CopyCompareLink):
(async CompareRedirect):
(FindCommit):
* Tools/Scripts/libraries/reporelaypy/reporelaypy/static/library/webkit.png: Added.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/templates/compare.html: Added.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/webserver.py:
(compare):

Canonical link: <a href="https://commits.webkit.org/289744@main">https://commits.webkit.org/289744@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/201982e48e3c8da1e666e08f569321819a14de13

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87433 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6946 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41805 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/92295 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/38172 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/89484 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/7328 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15111 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67536 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25274 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90435 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5583 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79117 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47876 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/86929 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5373 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33511 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/37288 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75791 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34374 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/94180 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14598 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10709 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76358 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14852 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74972 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75573 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18668 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19955 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18375 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/7536 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14616 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/14359 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17803 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16142 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

facetothefate commented on Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/search.js:
```diff
@@ -0,0 +1,153 @@
+async function GenerateCompareLink() {
+    const commitOne = document.getElementById('commit1').value;
+    const commitTwo = document.getElementById('commit2').value;
+    let message = document.getElementById('compareMessage');
```
should be 
``` javascript
const message
```

facetothefate commented on Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/search.js:
```diff
@@ -0,0 +1,153 @@
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+
+const COMMIT_URL = 'https://commits.webkit.org/';
```
It will be better that we use IIFE here, or we use ES module instead.

facetothefate commented on Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/search.js:
```diff
@@ -0,0 +1,153 @@
+    let message = document.getElementById('compareMessage');
+
+    if (!commitOne | !commitTwo) {
+        message.innerHTML = ERROR_TEMPLATE.replace("{message}", "Must provide two commits to compare");
```
Instead of using replace, you can make you template a function, pass in message.
```javascript
function errorMessage(message) {
    return `<div class="text small failed"> ${message} </div>`;
}
```

facetothefate commented on Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/search.js:
```diff
@@ -0,0 +1,153 @@
+const VALID_REF_RE = /^[a-zA-Z0-9\/\.\@-]+$/;
+const ERROR_TEMPLATE = `<div class="text small failed"> {message} </div>`;
+
+commitInputs = document.getElementsByClassName('commitInput');
```
```javascript
const commitInputs
```

facetothefate commented on Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/search.js:
```diff
@@ -0,0 +1,153 @@
+
+function FindCommit() {
+    const commit = document.getElementById('commitSearch').value;
+    let message = document.getElementById('searchMessage');
```
```javascript
const message
```

facetothefate commented on Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/search.js:
```diff
@@ -0,0 +1,153 @@
+        if (!response.ok) {
+            message.innerHTML = ERROR_TEMPLATE.replace("{message}", `Could not find ${commit}.<br>Please fix any typos or check if this commit exists.<br>`);
+        } else {
+            response.json().then(data => {
```
We should handle the request error

facetothefate replied:
response.json() may also have an error such like your API somehow didn't give a json back, and we should handle this as well

facetothefate reviewed (approved):
r+ with tiny comment

JonWBedard commented on Tools/Scripts/libraries/reporelaypy/reporelaypy/webserver.py:
```diff
@@ -76,6 +76,12 @@ def health():
+def compare():
+    return render_template('compare.html')
+
+
```
I don't see a landing page here. Won't we continue using the redirect?

briannafan replied:
Updated so this will be the landing page for commits.webkit.org!

briannafan committed:
commits.webkit.org should have a UI to inter-convert revision/hash/commit-identifier
https://bugs.webkit.org/show_bug.cgi?id=225540
rdar://78034339

Reviewed by Jonathan Bedard.

Introduces UI to commits.webkit.org!
Search and compare commits.

* Tools/Scripts/libraries/reporelaypy/reporelaypy/static/css/webkit.css: Added.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/main.js: Added.
(SetUpInputs):
* Tools/Scripts/libraries/reporelaypy/reporelaypy/static/js/search.js: Added.
(SetUpInputs):
(ErrorMessage):
(async GetCommit):
(ToggleButtons):
(ValidateCommitFormat):
(async GenerateCompareLink):
(async CopyCompareLink):
(async CompareRedirect):
(FindCommit):
* Tools/Scripts/libraries/reporelaypy/reporelaypy/static/library/webkit.png: Added.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/templates/compare.html: Added.
* Tools/Scripts/libraries/reporelaypy/reporelaypy/webserver.py:
(compare):

Canonical link: https://commits.webkit.org/289744@main
